### PR TITLE
Enhance/lu 2669 calculate imputation factors sg

### PR DESCRIPTION
**Pull request by GMReynolds**

### CODE REVIEW

**Comment by NigelBond** [comment]
This is mostly OK but there are a couple of things. Firstly there's some code that's commented out for testing - should this be in or out?

### COMMENTED OUT FOR TESTING ###
        # sqs.delete_message(QueueUrl=queue_url, ReceiptHandle=receipt_handle)

File: calculate_imputation_factors_wrangler.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+        # invoke the method to calculate the factors
+        calculate_factors = lambda_client.invoke(FunctionName=method_name, Payload=data_json)
+        json_response = calculate_factors.get('Payload').read().decode("UTF-8")
+
+        # send data to sqs queue
+        send_sqs_message(queue_url, json.loads(json_response), sqs_messageid_name)
+
+        ### COMMENTED OUT FOR TESTING ###
+        # sqs.delete_message(QueueUrl=queue_url, ReceiptHandle=receipt_handle)
```
Comment (NigelBond) [inline calculate_imputation_factors_wrangler.py]: This is commented out for testing. Should it be in or out?

File: calculate_imputation_factors_wrangler.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    arn = os.environ['arn']
+    checkpoint = os.environ['checkpoint']
+    sns = boto3.client('sns')
+
+    sns_message = {
+        "success": True,
+        "module": "Imputation Calculate Imputation Factors",
+        "checkpoint": checkpoint,
+        "anomalies": anomalies.to_json(orient='records'),
```
Comment (NigelBond) [inline calculate_imputation_factors_wrangler.py]: A reference to "anomalies". We had a discussion about this the other day - should it be removed?

**Comment by NigelBond** [comment]
Other than that it was fine. The Pep8 were passed with some minor quibbles

**Comment by NigelBond** [comment]
When running tests, we got a failure when trying to import the wrangler and method. We had to put a __init__.py file in the tests directory and this solved the issue. (NB the __init__.py file is empty, it just needs to be there)

**Comment by NigelBond** [comment]
After correcting the __init__.py thing, we got another error because the file 'tests/recalculate_means_output.json' was not in the test directory. So, this may need to be loaded into Git.

File: calculate_imputation_factors_wrangler.py (hunk 1 above)
Comment (GMReynolds) [reply calculate_imputation_factors_wrangler.py]: I have uncommented the code

File: calculate_imputation_factors_wrangler.py (hunk 2 above)
Comment (GMReynolds) [reply calculate_imputation_factors_wrangler.py]: anomalies have been removed from the sns function

**Comment by NigelBond** [comment]
All changes have been made to my approval

File: tests/test_calculate_imputation_factors.py [hunk 3]
```diff
@@ -87,98 +98,136 @@ def test_wrangler(self):
...
+        with mock.patch.dict(
+            calculate_imputation_factors_method.os.environ, {"queue_url": queue_url}
+        ):
+            calculate_imputation_factors_method.os.environ.pop("questions")
+            out = calculate_imputation_factors_method.lambda_handler(
+                {"RuntimeVariables": {"checkpoint": 666}}, None
+            )
+            self.assertRaises(ValueError)
+
+            assert out["error"].__contains__(
```
Comment (bitmonkey) [inline tests/test_calculate_imputation_factors.py]: Using 'magic methods' like this is a code smell...these are bound to python descriptors so you don't need to call it directly.

Your test should be `assert "ValueError: Error validating environment params:" in out["error"]`, underneath python converts `in` to calling the `__contains__` method so it's essentially the same.

**Comment by Dom-Ford** [comment]
@bitmonkey Changes made. please can you re-review.

**Pull request merged**